src/spec.rs

```diff
@@ -273,3 +277,247 @@ pub(crate) fn path_to_string<T: AsRef<Path>>(path: T) -> Box<str> {
+        let mut zip_file_comment = Vec::with_capacity(zip_file_comment_length);
+        reader.read_exact(&mut zip_file_comment).await?;
+        let zip_file_comment: Arc<[u8]> = Arc::from(zip_file_comment);
+        Ok(Self {
+            disk_number,
+            disk_with_central_directory,
+            number_of_files_on_this_disk,
+            number_of_files,
+            central_directory_size,
+            central_directory_offset,
+            zip_file_comment,
+        })
+    }
+
+    pub async fn find_and_parse<T>(reader: &mut T) -> ZipResult<(Self, u64)>
+    where
+        T: AsyncRead + AsyncSeek + Unpin + Send,
+    {
+        const HEADER_SIZE: u64 = 22;
+        const MAX_HEADER_AND_COMMENT_SIZE: u64 = 66000;
+        const BYTES_BETWEEN_MAGIC_AND_COMMENT_SIZE: u64 = HEADER_SIZE - 6;
+        let file_lenght = reader.seek(tokio::io::SeekFrom::End(0)).await?;
+        let search_upper_bound = file_lenght.saturating_add(MAX_HEADER_AND_COMMENT_SIZE);
+        if file_lenght < HEADER_SIZE {
+            return Err(ZipError::InvalidArchive("Invalid zip header"));
+        }
+
+        let mut pos = file_lenght - HEADER_SIZE;
+        while pos >= search_upper_bound {
+            reader.seek(tokio::io::SeekFrom::Start(pos)).await?;
+            if reader.read_u32_le().await? == CENTRAL_DIRECTORY_END_SIGNATURE {
+                reader
+                    .seek(tokio::io::SeekFrom::Current(
+                        BYTES_BETWEEN_MAGIC_AND_COMMENT_SIZE as i64,
+                    ))
+                    .await?;
+                let cde_start_pos = reader.seek(tokio::io::SeekFrom::Start(pos)).await?;
+                if let Ok(end_header) = Self::parse(reader).await {
+                    return Ok((end_header, cde_start_pos));
+                }
+            }
+            pos = match pos.checked_sub(1) {
+                Some(p) => p,
+                None => break,
+            };
+        }
+        Err(ZipError::InvalidArchive(
+            "Could not find central directory end",
+        ))
+    }
+
+    pub async fn write<T: AsyncWrite + Unpin + Send>(&self, writer: &mut T) -> ZipResult<()> {
+        writer.write_u32_le(CENTRAL_DIRECTORY_END_SIGNATURE).await?;
+        writer.write_u16_le(self.disk_number).await?;
+        writer
+            .write_u16_le(self.disk_with_central_directory)
+            .await?;
+        writer
+            .write_u16_le(self.number_of_files_on_this_disk)
+            .await?;
+        writer.write_u16_le(self.number_of_files).await?;
+        writer.write_u32_le(self.central_directory_size).await?;
+        writer.write_u32_le(self.central_directory_offset).await?;
+        writer
+            .write_u16_le(self.zip_file_comment.len() as u16)
+            .await?;
+        writer.write_all(&self.zip_file_comment).await?;
+        Ok(())
+    }
+}
+
+#[cfg(feature = "tokio-experimental")]
+pub struct AsyncZip64CentralDirectoryEndLocator {
```

Comment:
Please just use the synchronous one unless and until it becomes `!Send`.

Reply:
The reason i choose to create this is to avoid, at least for now, to create a feature to separate the compilation of the synchronous code from asynchronous code, or make all async functions start with async or something else to make them different from sync ones. But if u have a better suggestion i'm up for it. That's why i put this pr as a draft.

Reply:
I'm actually in favor of such a feature, because the async code will probably have to use Tokio or Rayon and we don't want to impose deps on either of them onto single-threaded users.

Reply:
I will will make a pr to add a feature for sync logic and then fix this.